addons/medical_gui/functions/fnc_updateActions.sqf

```diff
@@ -43,15 +43,17 @@ private _idcIndex = 0;
 {
     _x params ["_displayName", "_category", "_condition", "_statement"];
 
+    if (_forEachIndex < (GVAR(page)*9)) then {continue}; // Pagination
+
     // Check action category and condition
-    if (_category == _selectedCategory && {call _condition}) then {
-        private _ctrl = _display displayCtrl (IDCS_ACTION_BUTTONS select _idcIndex);
-        _ctrl ctrlSetText _displayName;
-        _ctrl ctrlShow true;
+    private _ctrl = _display displayCtrl (IDCS_ACTION_BUTTONS select _idcIndex);
+    _ctrl ctrlSetText _displayName;
+    _ctrl ctrlShow true;
+
+    _ctrl ctrlAddEventHandler ["ButtonClick", _statement];
+    _ctrl ctrlAddEventHandler ["ButtonClick", {GVAR(pendingReopen) = true}];
 
-        _ctrl ctrlAddEventHandler ["ButtonClick", _statement];
-        _ctrl ctrlAddEventHandler ["ButtonClick", {GVAR(pendingReopen) = true}];
+    _idcIndex = _idcIndex + 1;
+} forEach GVAR(possibleActions);
 
-        _idcIndex = _idcIndex + 1;
-    };
-} forEach GVAR(actions);
+if (_actionsCount > 0 && {_idcIndex isEqualTo 0}) then {GVAR(page) = 0;}; // Ensures pages all have actions
```

Comment:
Lazy eval will be slower with such simple condition. That comment does not make much sense for me, can you make it more descriptive?
```suggestion
if (_actionsCount > 0 && _idcIndex isEqualTo 0) then {GVAR(page) = 0}; // Ensures pages all have actions
```

Reply:
> Lazy eval will be slower with such simple condition.

**Been waiting for when someone says that.** Needs to be check the same for the whole mod, as there are lazy-evaled basic boolean checks, among many others, that would surely be much slower with lazy than without.

Reply:
Fixed lazy eval. See new comment, any unclarity? Effectively medical gui disables category being selected if no actions, so rather than a lot of logic simply check if no actions were added to the page when there's more than no actions available and put back to page 0 if so.